textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/regexmatch/GramBooleanQuery.java

```diff
@@ -293,6 +294,133 @@ private static String toLuceneQueryString(GramBooleanQuery query) {
 			}
 		}
 	}
+	
+	public boolean isEmpty() {
```

Comment:
For isEmpty(), I'd rather leave it public, because it kinda override the isEmpty function, just like how I override toString() in this class.